ld/ldlang.c

```diff
@@ -3686,10 +3686,16 @@ open_input_bfds (lang_statement_union_type *s, enum open_bfd_mode mode)
 	    plugin_insert = NULL;
 #endif
 	  break;
+         /* This is from a --defsym on the command line.  */
 	case lang_assignment_statement_enum:
-	  if (s->assignment_statement.exp->type.node_class != etree_assert)
-	    exp_fold_tree_no_dot (s->assignment_statement.exp);
-	  break;
+          if (s->assignment_statement.exp->type.node_class != etree_assert)
+          {
+          if(!(s->assignment_statement.exp->assign.defsym) && (s->assignment_statement.exp->type.node_class == etree_provide))
+          ;
+          else
+          exp_fold_tree_no_dot (s->assignment_statement.exp);
+          }
+          break;
```

Comment:
doesn't look microblaze related: to be inspected

Reply:
"improper address mapping of PROVIDE directive symbols[DTOR_END] are causing runtime loops and we don't need to override PROVIDE symbols if symbols are defined in libraries and linker so I am disabling override for PROVIDE symbols."

Reply:
Sounds like it fixes an issue. Keeping it in.

Reply:
Part of
https://github.com/Xilinx/meta-xilinx/blob/master/meta-microblaze/recipes-devtools/binutils/binutils/0027-Patch-MicroBlaze-improper-address-mapping-of-PROVIDE.patch
Depends on
https://github.com/Xilinx/meta-xilinx/blob/master/meta-microblaze/recipes-devtools/binutils/binutils/0021-Revert-ld-Remove-unused-expression-state.patch
Relates to https://github.com/zephyrproject-rtos/binutils-gdb/pull/10#discussion_r1421609298
Keeping it in.